src/rpa_gw.F

```diff
@@ -1389,6 +1391,23 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
          E_VBM_SCF_beta = -1.0E3
          E_CBM_SCF_beta = 1.0E3
 
+         ndos = 0
+         dos_precision = mp2_env%ri_g0w0%dos_prec
+         dos_upper_bound = mp2_env%ri_g0w0%dos_upper
+         dos_lower_bound = mp2_env%ri_g0w0%dos_lower
+
+         IF (dos_lower_bound >= dos_upper_bound) THEN
+            CALL cp_abort(__LOCATION__, "Invalid settings for GW_DOS calculation!")
+            dos_precision = 0_dp
```

Comment:
What is the effect of these follow-up variable initialisations after a cp_abort?